packages/simplr-forms-core/__tests__/stores/form-store.test.ts

```diff
@@ -150,4 +156,40 @@ describe("Form store", () => {
             done.fail(error);
         }
     });
+
+    it("registers field with props", () => {
+        const formId = "FORM-ID";
+        const fieldId = "FIELD-ID";
+        const fieldProps: MyFieldProps = {
+            name: "fieldName",
+            value: "initial-value"

```

Comment:
Add `randomKey` with `random value` for a non-default key: value situation check.